commons/src/main/java/com/powsybl/commons/parameters/DynamicValueParameter.java

```diff
@@ -0,0 +1,58 @@
+/**
+ * Copyright (c) 2024, All partners of the iTesla project (http://www.itesla-project.eu/consortium)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.commons.parameters;
+
+import com.powsybl.commons.config.ModuleConfig;
+
+import java.util.*;
+
+/**
+ * @author Antoine Bouhours {@literal <antoine.bouhours at rte-france.com>}
+ */
+public class DynamicValueParameter extends Parameter {
```

Comment:
For this enum to be consistent, I changed a bit the implementation to return a Dynamic Parameter when the default is overriden in the PlatformConfig and a Parameter otherwise. Before I always returned a Dynamic Parameter. See https://github.com/powsybl/powsybl-core/pull/3133/commits/a5b918bf0eba173b6dc56870952477a727fd5a00